File: 0128-longest-consecutive-sequence/solution.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        longest = 0
        nums_set = set(nums)

        for x in nums_set:
            if x - 1 not in nums_set:
                length = 1
                while x + length in nums_set:
                    length += 1
                longest = max(longest, length)

        return longest

File: 0128-longest-consecutive-sequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()){
            return 0;
        }
        int longest = 0;
        unordered_set<int> map(nums.begin(), nums.end());
        for (int cur:map){
            if (map.find(cur - 1) == map.end()){
                int count = 1;
                int temp = cur;
                while(map.count(temp + 1)){
                    count += 1;
                    temp += 1;
                } 
                longest = max(count, longest);
            }
        }
        return longest;

    }
};
